src/unix/apple/macosx/hid.c

```diff
@@ -164,15 +165,28 @@ struct hid *mty_hid_create(HID_CONNECT connect, HID_DISCONNECT disconnect, HID_R
 	CFArrayRef matches = CFArrayCreate(kCFAllocatorDefault, (const void **) dict_list, key ? 4 : 3, NULL);
 	IOHIDManagerSetDeviceMatchingMultiple(ctx->mgr, matches);
 
+	IOHIDManagerScheduleWithRunLoop(ctx->mgr, CFRunLoopGetCurrent(), kCFRunLoopDefaultMode);
+
+	IOReturn e = IOHIDManagerOpen(ctx->mgr, kIOHIDOptionsTypeNone);
+	if (e == kIOReturnExclusiveAccess && key) {
+		CFMutableDictionaryRef d4 = IOServiceNameMatching("org_pqrs_Karabiner_DriverKit_VirtualHIDKeyboard");
+		CFMutableDictionaryRef dict_list2[] = {d0, d1, d2, d4};
+
+		CFArrayRef matches2 = CFArrayCreate(kCFAllocatorDefault, (const void **) dict_list2, 4, NULL);
```

Comment:
it's a good habit to write `sizeof(dict_list2)/sizeof(dict_list2[0])` here instead of `4`. in this case the size is unlikely to change and the definition of the array is nearby, but it remains a good coding habit.